build/accelerate_launch.py

```diff
@@ -70,35 +69,40 @@ def main():
             if actions_type_map.get(key) == "_AppendAction":
                 for param_val in val:
                     accelerate_launch_args.extend([f"--{key}", str(param_val)])
+            elif (actions_type_map.get(key) == "_StoreTrueAction" and val) or (
+                actions_type_map.get(key) == "_StoreFalseAction" and not val
+            ):
+                accelerate_launch_args.append(f"--{key}")
             else:
                 accelerate_launch_args.append(f"--{key}")
                 # Only need to add key for params that aren't flags ie. --quiet
                 if actions_type_map.get(key) == "_StoreAction":
                     accelerate_launch_args.append(str(val))
 
-    if not accelerate_config.get("config_file"):
-        if json_configs.get("multi_gpu"):
-            # add FSDP config
+    num_processes = accelerate_config.get("num_processes")
+    if num_processes:
+        if num_processes > 1 and not accelerate_config.get("config_file"):
+            # Add default FSDP config
             fsdp_filepath = os.getenv(
                 "FSDP_DEFAULTS_FILE_PATH", "/app/accelerate_fsdp_defaults.yaml"
             )
             if os.path.exists(fsdp_filepath):
-                logging.info("Setting accelerate config file to: %s", fsdp_filepath)
-                accelerate_launch_args.append("--config_file")
-                accelerate_launch_args.append(fsdp_filepath)
-
-            # add num_processes to overwrite config file set one
-            if not accelerate_config.get("num_processes"):
-                num_gpus = torch.cuda.device_count()
-                if num_gpus > 1:
-                    logging.info("Setting accelerate num processes to %s", num_gpus)
-                    accelerate_launch_args.append("--num_processes")
-                    accelerate_launch_args.append(str(num_gpus))
-        else:
-            accelerate_launch_args.append("--num_processes")
-            accelerate_launch_args.append("1")
-
-    # add training_script
+                logging.info("Using accelerate config file: %s", fsdp_filepath)
+                accelerate_launch_args.extend(["--config_file", fsdp_filepath])
+
+        elif num_processes == 1:
+            logging.info(
+                "num_processes=1 so setting env var CUDA_VISIBLE_DEVICES=0: %s",
+                fsdp_filepath,
+            )
```

Comment:
no need to mention fsdp filepath
```suggestion
                "num_processes=1 so setting env var CUDA_VISIBLE_DEVICES=0",
```